task/task_service.go

```diff
@@ -173,7 +173,7 @@ func (ts *taskService) WaitForFinalStatus(taskID string, closer <-chan struct{})
 		select {
 		case <-closer:
 			return nil, nil
-		default:
```

Comment:
this fixes the fact that `WaitForFinalStatus` always blocked 1s which lead to the race condition with in the Create test